pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -153,25 +201,25 @@ func podMatchesAllAffinityTerms(terms []framework.AffinityTerm, pod *v1.Pod) boo
 //  1. Whether it has PodAntiAffinity
 //  2. Whether any AntiAffinityTerm matches the incoming pod
 func (pl *InterPodAffinity) getExistingAntiAffinityCounts(ctx context.Context, pod *v1.Pod, nsLabels labels.Set, nodes []*framework.NodeInfo) topologyToMatchedTermCount {
-	topoMaps := make([]topologyToMatchedTermCount, len(nodes))
+	antiAffinityCountsList := make([]topologyToMatchedTermCountList, len(nodes))
 	index := int32(-1)
 	processNode := func(i int) {
 		nodeInfo := nodes[i]
 		node := nodeInfo.Node()
 
-		topoMap := make(topologyToMatchedTermCount)
+		antiAffinityCounts := make(topologyToMatchedTermCountList, 0)
 		for _, existingPod := range nodeInfo.PodsWithRequiredAntiAffinity {
-			topoMap.updateWithAntiAffinityTerms(existingPod.RequiredAntiAffinityTerms, pod, nsLabels, node, 1)
+			antiAffinityCounts.appendWithAntiAffinityTerms(existingPod.RequiredAntiAffinityTerms, pod, nsLabels, node, 1)
 		}
-		if len(topoMap) != 0 {
-			topoMaps[atomic.AddInt32(&index, 1)] = topoMap
+		if len(antiAffinityCounts) != 0 {
+			antiAffinityCountsList[atomic.AddInt32(&index, 1)] = antiAffinityCounts
 		}
 	}
 	pl.parallelizer.Until(ctx, len(nodes), processNode, pl.Name())
 
 	result := make(topologyToMatchedTermCount)
 	for i := 0; i <= int(index); i++ {
```

Comment:
Since it's a list, we can iterate now over it instead. Same applies to getIncomingAffinityAntiAffinityCounts.

Reply:
It's iteration over antiAffinityCountsList, which don't need to be full (it has entries only when `antiAffinityCounts != 0` above). That's why we cap the iteration at `index`.

Reply:
I think you can still write:

`for tpc in antiAffinityCountsList[0:index]`